feat: CXSPA-4442 New components for email and personal info

scarai-sap opened the pull request:
ATT. Below is how it works now

Update new video with new css on 18:12 21 Sept. Please check it again



https://github.com/SAP/spartacus/assets/38693320/0aeccc2b-e62c-4855-a3bb-4b45ea484102

scarai-sap committed:
CXSPA-4442: 1st version

github-actions[bot] committed:
Add license header

scarai-sap committed:
CXSPA-4442: style fixing

scarai-sap committed:
CXSPA-4442: style fixing

scarai-sap committed:
Merge branch 'feature/CXSPA-4442' of https://github.com/SAP/spartacus into feature/CXSPA-4442

scarai-sap committed:
CXSPA-4442: adding msgstrips

scarai-sap committed:
CXSPA-4442: adding UT for msgstrips

scarai-sap committed:
CXSPA-4442: adding UT for msgstrips

scarai-sap committed:
Merge branch 'develop-6.5.x' into feature/CXSPA-4442

evy-yang01 commented:
Button is supposed to be align to the right consistent with the input right place according to the mockup.
<img width="739" alt="image" src="https://github.com/SAP/spartacus/assets/51769930/5623103a-c135-46f0-b15f-deb5284d45cd">

evy-yang01 commented:
One question not very sure, don't we need to add our index to public_api.ts in components folder?
<img width="1019" alt="image" src="https://github.com/SAP/spartacus/assets/51769930/d4fc5377-16ed-4496-a2f6-f5f2335ea8e4">

scarai-sap commented:
> Button is supposed to be align to the right consistent with the input right place according to the mockup. <img alt="image" width="739" src="https://user-images.githubusercontent.com/51769930/268885535-5623103a-c135-46f0-b15f-deb5284d45cd.png">

I guess you want to align to left?

scarai-sap commented:
> Button is supposed to be align to the right consistent with the input right place according to the mockup. <img alt="image" width="739" src="https://user-images.githubusercontent.com/51769930/268885535-5623103a-c135-46f0-b15f-deb5284d45cd.png">

In fact, it aligned and the padding is introduced by button.scss :(
![image](https://github.com/SAP/spartacus/assets/38693320/1064a099-e773-4fbc-8cb9-e66de84a1763)

scarai-sap commented:
> public_api.ts

Be frankly, I don't know why we need it. I guess you are right. We need add it as other component does

i53577 commented on feature-libs/user/profile/styles/_new-email.scss:
```diff
@@ -0,0 +1,63 @@
+%cx-new-email {
+  .email-enhancedUI-header-area {
+    border: 1px, 0px, 0px, 0px;
```
maybe use rem instead of px?

i53577 commented on feature-libs/user/profile/styles/_new-email.scss:
```diff
@@ -0,0 +1,63 @@
+  }
+
+  .email-enhancedUI-body-area {
+    border: 0px, 0px, 1px, 0px;
```
maybe use rem instead of px?

i53577 commented on feature-libs/user/profile/styles/_new-profile.scss:
```diff
@@ -0,0 +1,48 @@
+%cx-new-profile {
+  .myaccount-enhancedUI-header-area {
+    border: 1px, 0px, 0px, 0px;
```
maybe use rem instead of px?

scarai-sap replied:
removed

i53577 commented on feature-libs/user/profile/styles/_new-profile.scss:
```diff
@@ -0,0 +1,48 @@
+  }
+
+  .myaccount-enhancedUI-body-area {
+    border: 0px, 0px, 1px, 0px;
```
maybe use rem instead of px?

scarai-sap replied:
removed

i53577 commented on feature-libs/user/profile/components/new-user-profile/new-profile.component.spec.ts:
```diff
@@ -0,0 +1,169 @@
+          {
+            provide: FeaturesConfig,
+            useValue: {
+              features: { level: '5.2' },
```
maybe config it as 6.6 or * ?

scarai-sap replied:
Ah, Just cp from somewhere. Will delete it

scarai-sap committed:
CXSPA-4442: update code after code review

scarai-sap committed:
CXSPA-4442: fix styling

scarai-sap committed:
Merge branch 'feature/CXSPA-4442' of https://github.com/SAP/spartacus into feature/CXSPA-4442

scarai-sap committed:
Empty-Commit

scarai-sap committed:
CXSPA-4442: fix styling

scarai-sap committed:
Merge branch 'develop-6.5.x' into feature/CXSPA-4442

scarai-sap committed:
Merge branch 'feature/CXSPA-4442' of https://github.com/SAP/spartacus into feature/CXSPA-4442

scarai-sap committed:
update code to v2

scarai-sap committed:
CXSPA-4442: fix issue

scarai-sap committed:
CXSPA-4442: Create new combined component

scarai-sap committed:
CXSPA-4442: For yarn prettier:fix

github-actions[bot] committed:
Add license header

scarai-sap committed:
CXSPA-4442: For UT

scarai-sap committed:
CXSPA-4442: update css

scarai-sap committed:
Merge branch 'feature/CXSPA-4442' of https://github.com/SAP/spartacus into feature/CXSPA-4442

scarai-sap committed:
Merge branch 'develop-6.5.x' into feature/CXSPA-4442

i53577 commented on feature-libs/user/profile/styles/_new-combine-profile.scss:
```diff
@@ -0,0 +1,9 @@
+%cx-new-combined-profile {
+  .new-combined-profile-container {
+    padding-bottom: 5rem;
+    border-width: 1px 0px 1px 0px;
```
Oops, does this border really take effect? `px` is used here

scarai-sap replied:
Yes, you can check the video

i53577 replied:
Looks like this is for the top and bottom line. In this case, there are 2 overlapping lines between 2 sections. And from the video, that overlapping line looks bold than others.

scarai-sap replied:
Fixed, Please check below snapshot
<img width="1347" alt="image" src="https://github.com/SAP/spartacus/assets/38693320/ffe094ec-a6ef-471b-b744-4dcf738b509b">

evy-yang01 commented:
@scarai-sap , seems UX asked for the space between input box and save button group. Could you add it.
<img width="629" alt="image" src="https://github.com/SAP/spartacus/assets/51769930/636d16b7-e577-4e5c-b09a-ead94154957f">


`  .myaccount-label-padding {
    padding-bottom: 0.75rem !important;
  }`

scarai-sap commented:
> @scarai-sap , seems UX asked for the space between input box and save button group. Could you add it. <img alt="image" width="629" src="https://user-images.githubusercontent.com/51769930/269549974-636d16b7-e577-4e5c-b09a-ead94154957f.png">
> 
> ` .myaccount-label-padding { padding-bottom: 0.75rem !important; }`

I added as below
  .email-enhancedUI-button {
    font-weight: var(--cx-font-weight-normal);
    padding: 0.8rem, 1.5rem, 0.8rem, 1.5rem;
    font-size: $font-size-base;
  }

scarai-sap committed:
CXSPA-4442: adding padding as code reviewer suggested

scarai-sap committed:
Merge branch 'feature/CXSPA-4442' of https://github.com/SAP/spartacus into feature/CXSPA-4442

scarai-sap committed:
CXSPA-4442: adding padding as code reviewer v2

scarai-sap committed:
CXSPA-4442:  Adjust edit button

scarai-sap committed:
Merge branch 'develop-6.5.x' into feature/CXSPA-4442

scarai-sap committed:
CXSPA-4442:  Adjust line

scarai-sap committed:
Merge branch 'feature/CXSPA-4442' of https://github.com/SAP/spartacus into feature/CXSPA-4442

scarai-sap reviewed (commented):
Latest display UI as below
<img width="1347" alt="image" src="https://github.com/SAP/spartacus/assets/38693320/e80781a8-22a5-4a62-8b86-b8605495be83">

scarai-sap committed:
rerun pp

i53577 reviewed (dismissed):
LGTM

scarai-sap committed:
trigger pp again

giancorderoortiz committed:
Merge branch 'develop-6.5.x' into feature/CXSPA-4442

anjana-bl committed:
bug fix: CDC installation Issue (#17898)

rmch91 committed:
fix: add FeaturesConfigModule to CheckoutReviewSubmitModule (#17921)

FeaturesConfigModule has been added to CheckoutReviewSubmitModule to get rid of error

kpawelczak committed:
chore: release 6.5.1 (#17930)

scarai-sap committed:
Merge branch 'develop-6.5.x' into feature/CXSPA-4442

scarai-sap committed:
Merge branch 'develop-6.7.x' of https://github.com/SAP/spartacus into feature/CXSPA-4442

scarai-sap committed:
rename v1

scarai-sap committed:
rename v2

scarai-sap committed:
CXSPA-4442: update code with rename and test

scarai-sap committed:
CXSPA-4442: Fix issue for i18n

scarai-sap committed:
CXSPA-4442: Fix issue can;'t display properly

scarai-sap committed:
CXSPA-4442: use USE_MY_ACCOUNT_V2_PROFILE and USE_MY_ACCOUNT_V2_EMAIL

scarai-sap committed:
CXSPA-4442: Fix format

github-actions[bot] committed:
Add license header

scarai-sap committed:
CXSPA-4442: Fix format

scarai-sap committed:
Merge branch 'feature/CXSPA-4442' of https://github.com/SAP/spartacus into feature/CXSPA-4442

scarai-sap committed:
CXSPA-4442: make feature available in lib level

scarai-sap committed:
Merge branch 'develop-6.7.x' into feature/CXSPA-4442

scarai-sap committed:
test for update dependencies.json

scarai-sap committed:
CXSPA-4442: Add IT and bug fixing

scarai-sap committed:
update code with rename relocate and fix css issue

scarai-sap committed:
format prettier

scarai-sap committed:
Merge branch 'develop-6.7.x' into feature/CXSPA-4442

scarai-sap committed:
CXSPA-4442: Fix typo

i53577 commented:
please add accessibility test

scarai-sap commented:
> please add accessibility test

let me check whether it's in the scope

scarai-sap commented:
> please add accessibility test

it's in https://jira.tools.sap/browse/CXSPA-4449

i53577 commented:
Your e2e tests are already here. So I am curious why accessibility test is not. But it's ok for me to use another ticket that you mentioned to track it.

i53577 commented on projects/storefrontapp-e2e-cypress/cypress/helpers/update-email.ts:
```diff
@@ -20,7 +20,7 @@ export function registerAndLogin() {
   it('should update his email address and login', () => {
     const newUid = generateMail(randomString(), true);
-    cy.get('cx-update-email').within(() => {
+    cy.get('cx-update-email, cx-my-new-account-v2-email').within(() => {
```
if v1 and v2 will exist at the same time, then does this change mean v1's tests will not pass in the future?

scarai-sap replied:
this grammar means A or B in here.

i53577 commented on projects/storefrontapp-e2e-cypress/cypress/helpers/update-profile.ts:
```diff
@@ -23,12 +23,19 @@ export function updateProfile(user?: SampleUser) {
     );
     cy.get('[formcontrolname="lastName"]').should('have.value', user.lastName);
   }
+  cy.get('.myaccount-enhancedUI-editButton')
```
maybe it's better to add new functions instead of modifying existing ones that for v1.

scarai-sap replied:
two sides of the coin,  modify/reuse code make it easily if we wanna change sth in the future

i53577 replied:
I don't get the point. Does V1 also have this element `myaccount-enhancedUI-editButton`? Won't it cause any failure in  V1‘s tests?

scarai-sap replied:
if no this element, not click edit. ===> in V1, it loaded with edit model
if with this element. click edit ==> n V2, transfer to edit model.

Then we can reuse edit feature as it's totally same.

scarai-sap committed:
Merge branch 'develop-6.7.x' into feature/CXSPA-4442

scarai-sap committed:
Fix issue reported by QA

scarai-sap committed:
CXSPA-5564: fix long name and email
